portals/publisher/src/main/webapp/source/src/app/components/GlobalPolicies/Listing/Listing.tsx

```diff
@@ -0,0 +1,1115 @@
+    id: string;
+    name: string;
+    displayName: string;
+    type: string;  
+    serverUrl: string;
+    provider: string;
+    showInApiConsole: boolean;
+    vhosts: any;
+    endpointURIs: any;
+    additionalProperties: any;
+}
+
+interface SelectedGateway {
+    id: string;
+    gatewayLabels: string[];
+}
+
+/**
+ * Global policies Lisitng Page.
+ * @returns {TSX} - Listing Page.
+ */
+const Listing: React.FC = () => {
+    const classes = useStyles();
+    const [policies, setPolicies] = useState<Policy[]>([]);
+    const [selectedGateways, setSelectedGateways] = useState<SelectedGateway[]>([]);
+    const [environments, setEnvironments] = useState<Environment[]>([]);
+    const [loading, setLoading] = useState(false);
+    const [notFound, setnotFound] = useState(false);
+    const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+    const [isDeployDialogOpen, setIsDeployDialogOpen] = useState(false);
+    const [isUndeployDialogOpen, setIsUndeployDialogOpen] = useState(false);
+    const [selectedPolicyName, setSelectedPolicyName] = useState('');
+    const [selectedPolicyId, setSelectedPolicyId] = useState('');
+    const [deployingGateway, setDeployingGateway] = useState('');
+    const theme : any = useTheme();
+    const { globalPolicyAddIcon } = theme.custom.landingPage.icons;
+    const intl = useIntl();
+
+    /**
+     * Empty array to store the policies
+     * @param {Policy[]} inputPolicies - Policies.
+     * @returns {Policy[]} - Initial object array to hold selected gateways from the dropdown.
+     */
+    const getInitialSelectedGateways = (inputPolicies: Policy[]): SelectedGateway[] => {
+        const selectedGatewaysList: SelectedGateway[] = [];
+    
+        inputPolicies.forEach((policy) => {
+            const { id } = policy;
+            const selectedGatewayValue: SelectedGateway = {
+                id,
+                gatewayLabels: [],
+            };
+            selectedGatewaysList.push(selectedGatewayValue);
+        });
+
+        return selectedGatewaysList;
+    };
+
+    /**
+     * After deploying or undeploying, we need to clean the selected gateways list.
+     * @param {string} policyId - Policy ID.
+     * @returns {SelectedGateway[]} - Deployed policies' selected gateways should be empty. 
+     */
+    const cleanSeletectedGateways = (policyId: string): SelectedGateway[] => {
+        const updatedSelectedGateways: SelectedGateway[] = selectedGateways.map((data) => {
+            if (data.id === policyId) {
+                return {
+                    ...data,
+                    gatewayLabels: [],
+                };
+            }
+            return data;
+        });
+        
+        return updatedSelectedGateways;
+    };
+
+    /**
+     * Fetch the data from the backend to the compoenent.
+     */
+    const fetchGlobalPolicies = () => {
+        setLoading(true);
+        // Due to a bug in current backend, we pass 0, 10 as a workaround for now
```

Comment:
Is this bug fixed now?

Reply:
This is now fixed and I forgot to remove the comment. Addressed the required changes by [fddf57f](https://github.com/wso2/apim-apps/pull/515/commits/fddf57fa7308a53094c808e9bf2d2b40ed71b644) and [0ad490c](https://github.com/wso2/apim-apps/pull/515/commits/0ad490c497b08f999babbe521feaf83d029a6698)